Content.Server/Mobs/Systems/MobsSystem.cs

```diff
@@ -0,0 +1,68 @@
+using Content.Server.Chat.Systems;
+using Content.Shared.Mobs;
+using Content.Shared.Speech.Muting;
+
+namespace Content.Server.Mobs.Systems;
+
+public sealed class MobsSystem : EntitySystem
+{
+    [Dependency] private readonly ChatSystem _chat = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<DeathgaspComponent, MobStateChangedEvent>(OnDeath);
+        SubscribeLocalEvent<UnconsciousComponent, MobStateChangedEvent>(OnUnconscious);
+    }
+
+    private void OnDeath(EntityUid uid, DeathgaspComponent component, MobStateChangedEvent args)
+    {
+        // don't deathgasp if they arent going straight from crit to dead
+        if (args.NewMobState != MobState.Dead || args.OldMobState != MobState.Critical)
+            return;
+
+        Deathgasp(uid, component);
+    }
+
+    private void OnUnconscious(EntityUid uid, UnconsciousComponent component, MobStateChangedEvent args)
+    {
+        // don't unconscious if they arent going straight from alive to crit
+        if (args.NewMobState != MobState.Critical || args.OldMobState != MobState.Alive)
+            return;
+
+        Unconscious(uid, component);
+    }
+
+    /// <summary>
+    ///     Causes an entity to perform their deathgasp emote, if they have one.
+    /// </summary>
+    public bool Deathgasp(EntityUid uid, DeathgaspComponent? component = null)
+    {
+        if (!Resolve(uid, ref component, false))
+            return false;
+
+        if (HasComp<MutedComponent>(uid))
+            return false;
+
+        _chat.TryEmoteWithChat(uid, component.Prototype, ignoreActionBlocker: true);
+
+        return true;
+    }
```

Comment:
I'm pretty sure it's the style now to merge the component and the EntityUid into one parameter (https://docs.spacestation14.com/en/general-development/codebase-info/conventions.html#public-api-method-signature).

As you're already making a breaking change by renaming the system, it's probably fine to change the method signatures to be more up to date with code-base conventions.

```suggestion
    public bool Deathgasp(Entity<DeathgaspComponent?> ent)
    {
        if (!Resolve(ent, ref ent.Comp, false))
            return false;

        if (HasComp<MutedComponent>(ent))
            return false;

        _chat.TryEmoteWithChat(ent, ent.Comp.Prototype, ignoreActionBlocker: true);

        return true;
    }
```

You'll also need to change the calls to `Deathgasp((uid, component))` if you resolve it before passing it in.

(Note that I'm not a maintainer, and I'm not 100% sure what's with the transition in style.)

Reply:
Tbh I seen both approaches all over the codebase, so, idk.

Reply:
From my understanding, the old approach is "on the way out."

The problem is that it's so entrenched that it'll take years at the current pace for all of them to be updated... (Plus it causes a public API change, which we generally want to avoid.)